src/App.ts

```diff
@@ -1056,6 +1082,25 @@ export default class App<AppCustomContext extends StringIndexed = StringIndexed>
     let { client } = this;
     const token = selectToken(context);
 
+    // TODO :: WorkflowFunction owns these same utilities. Rework TS to allow for reuse.
+    // Set complete() and fail() utilities for function-related interactivity
+    if (type === IncomingEventType.Action && context.functionExecutionId !== undefined) {
+      listenerArgs.complete = (params: Parameters<FunctionCompleteFn>[0] = {}) => client.functions.completeSuccess({
+        token: context.functionBotToken,
```

Comment:
I there a specific reason we want to make this request with the `functionBotToken`?
Their may be a potential use case with OAuth where user tokens found in the client could be used to do this (I have not confirmed it yet)

Reply:
Specifically for the function complete/fail APIs, I think we should always use the function / JIT token. For some of the other helper / convenience functions, like `say` for posting a message, indeed we may want to think about what the API surface should look like in the case where the user may want to use a specific token (like user tokens for OAuth-enabled apps) - good call Will!

Reply:
Good callout. Addressing the point

> we should always use the function / JIT token

Will's point reminds me that we still need to honor the opt-out'ability (`attachFunctionToken`); forcing the JIT token to be used here regardless of opt-out doesn't honor that contract. 

So, I assume we should defer to their preferred token in utilities, as well.

Reply:
`token: context.functionBotAccessToken || token,` will attach the JIT token if the developer has not opted-out of automatic usage, and fall back to the usual, already-existing token logic if so (`botToken` first, followed by `userToken` if that doesn't exist).